packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -245,12 +276,14 @@ function getGraphPositions<T>({
   positions,
   heights,
   whitespaceThreshold,
+  columnSpan,
   ...commonGetPositionArgs
 }: {
```

Comment:
another performance improvement: instead of using deep recursion, we can do the same thing with a queue. that way there is less overhead in the stack so we can do faster computations:
```
function getGraphPositions<T>({
  items,
  positions,
  heights,
  whitespaceThreshold,
  columnSpan,
  ...commonGetPositionArgs
}: {
  items: Array<T>,
  heights: Array<number>,
  positions: Array<{ item: T, position: Position }>,
  whitespaceThreshold?: number,
  columnSpan: number,
  centerOffset: number,
  columnWidth: number,
  columnWidthAndGutter: number,
  gutter: number,
  measurementCache: Cache<T, number>,
  positionCache?: Cache<T, Position>,
}): {
  winningNode: NodeData<T>,
  additionalWhitespace: number,
} {
  const graph = new Graph<T>();
  const queue = [{ nodeData: { id: 'start', heights, positions }, index: 0 }];
  let bestNode = { nodeData: queue[0].nodeData, score: Infinity };

  while (queue.length > 0) {
    const { nodeData, index } = queue.shift();
    const currentPositions = nodeData.positions;
    const currentHeights = nodeData.heights;

    for (let i = index; i < items.length; i++) {
      const item = items[i];
      const itemHeight = measurementCache.get(item);
      const { top, left, updatedHeights } = updateColumnHeightsAndPositions({
        centerOffset: commonGetPositionArgs.centerOffset,
        columnWidthAndGutter: commonGetPositionArgs.columnWidthAndGutter,
        gutter: commonGetPositionArgs.gutter,
        heights: currentHeights,
        columnStartIndex: mindex(currentHeights), // mindex finds the minimum index
        columnSpan: 1, // For single-column items
        itemHeight,
      });

      const newPosition = { item, position: { top, left, width: commonGetPositionArgs.columnWidth, height: itemHeight } };
      const newPositions = [...currentPositions, newPosition];
      const newNodeData = { id: item, heights: updatedHeights, positions: newPositions };
      graph.addNode(newNodeData);
      graph.addEdge(nodeData, newNodeData, Math.min(...updatedHeights));

      if (whitespaceThreshold && Math.min(...updatedHeights) < whitespaceThreshold) {
        if (Math.min(...updatedHeights) < bestNode.score) {
          bestNode = { nodeData: newNodeData, score: Math.min(...updatedHeights) };
        }
        break; // Break early if a good enough layout is found
      }

      queue.push({ nodeData: newNodeData, index: i + 1 });
    }
  }

  return bestNode.nodeData.positions.length > 0 ? { winningNode: bestNode.nodeData, additionalWhitespace: bestNode.score } : { winningNode: graph.findLowestScoreNode('start'), additionalWhitespace: Infinity };
}

```

Reply:
Interesting, I could try this but since its a bigger change I would like to test it independently